src/transaction/transaction_single_node_group_complete_manager.cpp

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (C) 2008 Search Solution Corporation. All rights reserved by Search Solution.
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ */
+
+//
+// Manager of completed group on single node
+//
+
+#include "log_manager.h"
+#include "thread_manager.hpp"
+#include "transaction_single_node_group_complete_manager.hpp"
+
+namespace cubtx
+{
+  single_node_group_complete_manager::~single_node_group_complete_manager ()
+  {
+  }
+
+  //
+  // get global single node instance
+  //
+  single_node_group_complete_manager *single_node_group_complete_manager::get_instance ()
+  {
+    if (gl_single_node_group == NULL)
+      {
+	gl_single_node_group = new single_node_group_complete_manager ();
+      }
+    return gl_single_node_group;
+  }
+
+  //
+  // init initialize single node group commit
+  //
+  void single_node_group_complete_manager::init ()
+  {
+    cubthread::looper looper = cubthread::looper (std::chrono::milliseconds (10));
```

Comment:
Do we want to wait 10 miliseconds before creating a group ?
If this is not final behavior, a TODO will clarify this aspect